### DISCUSSION TITLE
Enable Form Templates for PR description and comment using JSON schemas

### DISCUSSION THREAD

**Discussion by AllanOricil** [community/community#55333] [Pull Requests]
There are certain pipelines/actions/automations that are dynamic based on inputs found in PR descriptions/comments. 

With some sort of advanced form feature, developers would not make mistakes when opening/updating their PRs description/comments.

Currently, when the PR description is wrong, developers have to wait until the job/automation tells them that the configuration can't be parsed. But if there was some sort of form validation, this would not happen, and we would avoid spending computing resources just to know that the configuration is wrong.

Please, implement this feature using JSON schemas instead of that horrible issues template sintax. A  json schema with a given name would have to be added to `.github` folder.

## Before

PR is opened/updated -> Job runs -> exception is thrown (description/comment is wrong) [few actions minutes doing nothing]

## After

PR is opened/updated -> [pr form validation validate all inputs] -> if everything is fine, job starts

A new object would also be added to PR events. This way developers can use expressions to read the form data as follows `{{ fromJSON(github.event.body_form) }}` in their github actions.

````json
{
    ...
    "body_form":{
         "input":  "foo"
    }
    ...
}

````

**Comment by JekaHFMN100** [comment]
> There are certain pipelines/actions/automations that are dynamic based on inputs found in PR descriptions/comments. 
> 
> With some sort of advanced form feature, developers would not make mistakes when opening/updating their PRs description/comments.
> 
> Currently, when the PR description is wrong, developers have to wait until the job/automation tells them that the configuration can't be parsed. But if there was some sort of form validation, this would not happen, and we would avoid spending computing resources just to know that the configuration is wrong.
> 
> Please, implement this feature using JSON schemas instead of that horrible issues template sintax. A  json schema with a given name would have to be added to `.github` folder.
> 
> 
> ## Before
> 
> PR is opened/updated -> Job runs -> exception is thrown (description/comment is wrong) [few actions minutes doing nothing]
> 
> ## After
> 
> PR is opened/updated -> [pr form validation validate all inputs] -> if everything is fine, job starts
> 
> 
> 
> 
> A new object would also be added to PR events. This way developers can use expressions to read the form data as follows `{{ fromJSON(github.event.body_form) }}` in their github actions.
> 
> ````json
> {
>     ...
>     "body_form":{
>          "input":  "foo"
>     }
>     ...
> }
> 
> ````

**Comment by Marvin-Brouwer** [comment]
When you're team is large enough, you will also run into situations where people just tend to forget reading the markdown template.
Having a form template would save some "policing" time for other team members.

**Comment by Marvin-Brouwer** [comment]
Hi,

Just revisiting this suggestion to see if it’s still under consideration.
It would be really helpful to have this feature.

Any updates or feedback from the GitHub team would be greatly appreciated!
